src/api/services/publish.js

```diff
@@ -17,19 +18,19 @@ export default async ctx => {
         [SCOPE_COLLECTION, SCOPE_DOCUMENT].includes(c.scope),
     );
 
-    await clearPublished(ctx);
-    await ctx.publishDocuments(ctx, count, collectionScopeFields);
+    (await isJobActive(ctx)) && (await clearPublished(ctx));
```

Comment:
Why is isJobActive called for every line? Can we make it a big if on top? Or is it meant to be able to stop at any step of the publication?
Also, there's no return condition, which means if it's not active it'll try all the subsequent calls? and finish anyway?

Reply:
If isJobActive should legit be called for every operation, then make it a bit more readable with a function like "chainOperationsWhileJobActive" ?